applications/affine-finetune-embeddings/model.py

```diff
@@ -0,0 +1,122 @@
+import torch
+import torch.nn.functional as F
+import pytorch_lightning as pl
+import torchmetrics
+from transformers import AutoModel, AutoTokenizer
+
+embedding_size = 384
+
+
+# Similarity Model
+class SimilarityModel(pl.LightningModule):
+    def __init__(self, n_dims, dropout_fraction, lr, use_relu):
+        super(SimilarityModel, self).__init__()
+        self.tokenizer = AutoTokenizer.from_pretrained("BAAI/bge-small-en-v1.5")
+        self.embedding_model = AutoModel.from_pretrained("BAAI/bge-small-en-v1.5")
+        for param in self.embedding_model.parameters():
+            param.requires_grad = False
+        self.matrix = torch.nn.Parameter(
+            torch.rand(
+                embedding_size,
+                n_dims,
+            )
+        )
+        torch.nn.init.xavier_uniform_(self.matrix)
+        self.dropout_fraction = dropout_fraction
+        self.lr = lr
+        self.use_relu = use_relu
+
+        self.recall = torchmetrics.Recall(task="binary")
+        self.f1 = torchmetrics.F1Score(num_classes=2, task="binary")
+        self.precision = torchmetrics.Precision(task="binary")
+        self.acc = torchmetrics.Accuracy(task="binary")
+        self.auc = torchmetrics.AUROC(task="binary")
+        self.save_hyperparameters()
+
+    def forward(self, text1, text2):
```

Comment:
this is vectorized right? 

List[str]?

Reply:
Going to modify the forward pass to take in a List (or maybe tensor) of tokenized texts. This is how other embedding models like bge do it and it should make training step more efficient since all the texts passed into batches will be pre-tokenized before training.